bindings/java/fdbJNI.cpp

```diff
@@ -1474,25 +1474,47 @@ JNIEXPORT jlong JNICALL Java_com_apple_foundationdb_FDBTransaction_Transaction_1
 			throwRuntimeEx(jenv, "Error getting handle to native resources");
 		return 0;
 	}
+	FDBTransaction* tr = (FDBTransaction*)tPtr;
+	FDBFuture* f;
+	if (matchIndex == -1) {
```

Comment:
not sure if there is better way to do this, we can also differentiate it at .java and have 2 JNI methods. Need to add test at java side.